[18.0][MIG] auth_jwt: Migration to 18.0

dnplkndll opened the pull request.

sbidoul committed:
[ADD] auth_jwt

sbidoul committed:
auth_jwt: use PyJWT instead of python-jose

Because it allows validating with a list of audiences.

sbidoul committed:
auth_jwt: add signature algorithms

sbidoul committed:
auth_jwt: support multiple audiences

sbidoul committed:
auth_jwt: add nbf validation test

sbidoul committed:
auth_jwt: docs clarification and fixes

sbidoul committed:
auth_jwt: fix jwks URI support

Make it work with pyjwt.

sbidoul committed:
[MIG] auth_jwt

sbidoul committed:
auth_jwt: mock instead of committing in tests

sbidoul committed:
auth_jwt: more precise precondition check

sbidoul committed:
Rename auth_jwt_test to auth_jwt_demo

oca-travis committed:
[UPD] Update auth_jwt.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auth_jwt 14.0.1.0.1

sbidoul committed:
[IMP] auth_jwt: add public_or_jwt auth method

This method is useful for public endpoints that need
to work for anonymous user, but can be enhanced when
an authenticated user is know.

A typical use case is a "add to cart" enpoint that can
work for anonymous users, but can be enhanced by
binding the cart to a known customer when the authenticated
user is known.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auth_jwt 14.0.1.1.0

yankinmax committed:
auth_jwt: Relicence under LGPL

OCA-git-bot committed:
auth_jwt 14.0.1.2.0

paradoxxxzero committed:
[IMP] auth_jwt: Add validator.next_validator_id to allow validator chaining

Commit:
[UPD] Update auth_jwt.pot

OCA-git-bot committed:
auth_jwt 14.0.2.0.0

sbidoul committed:
[MIG] auth_jwt from 14 to 16

sbidoul committed:
[MIG] auth_jwt: convert unit tests to integration tests

The unit tests were broken for non-functional reasons (interaction with
the mock) and is easier to implement as integration test.

Commit:
[UPD] Update auth_jwt.pot

OCA-git-bot committed:
[UPD] README.rst

sbidoul committed:
auth_jwt: add cookie mode

sbidoul committed:
auth_jwt: clarify exceptions

Distinguish errors that lead to a 401
from internal configuration errors.

sbidoul committed:
auth_jwt: minor refactoring

sbidoul committed:
[IMP] auth_jwt: refactor

Extract _parse_bearer_authorization function for easier reuse by fastapi_auth_jwt

sbidoul committed:
[FIX] auth_jwt: don't use public mode if a cookie is present

sbidoul committed:
[IMP] auth_jwt: check cookie_name is present in cookie mode

Commit:
[UPD] Update auth_jwt.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
auth_jwt 16.0.1.1.0

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (64 of 64 strings)

Translation: server-auth-16.0/server-auth-16.0-auth_jwt
Translate-URL: https://translation.odoo-community.org/projects/server-auth-16-0/server-auth-16-0-auth_jwt/es/

OCA-git-bot committed:
[UPD] README.rst

rbellanova committed:
Added translation using Weblate (Italian)

rbellanova committed:
Translated using Weblate (Italian)

Currently translated at 89.0% (57 of 64 strings)

Translation: server-auth-16.0/server-auth-16.0-auth_jwt
Translate-URL: https://translation.odoo-community.org/projects/server-auth-16-0/server-auth-16-0-auth_jwt/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (64 of 64 strings)

Translation: server-auth-16.0/server-auth-16.0-auth_jwt
Translate-URL: https://translation.odoo-community.org/projects/server-auth-16-0/server-auth-16-0-auth_jwt/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (64 of 64 strings)

Translation: server-auth-16.0/server-auth-16.0-auth_jwt
Translate-URL: https://translation.odoo-community.org/projects/server-auth-16-0/server-auth-16-0-auth_jwt/it/

MikeAelbrecht committed:
[MIG] auth_jwt: Migration to 17.0

dnplkndll committed:
[MIG] auth_jwt: Migration to 18.0

lmignon commented:
/ocabot migration auth_jwt

Commit:
[UPD] Update auth_jwt.pot

OCA-git-bot committed:
[BOT] post-merge updates

sbidoul commented:
Thanks for working on this. Could you re-do the migration based on the merged 17.0 version? Your migration commit is ok, but the history of the 17.0 branch you started with is a bit messy and has been cleaned-up since.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-auth-17.0/server-auth-17.0-auth_jwt
Translate-URL: https://translation.odoo-community.org/projects/server-auth-17-0/server-auth-17-0-auth_jwt/

dnplkndll committed:
[IMP] auth_jwt: pre-commit auto fixes

kobros-tech reviewed (approved):
@lmignon 
@sbidoul 

could you review this PR, we have merged migration for 17.0 and we can migrate it for 18.0

ChuwenSun reviewed (approved):
Looks good to me.

dnplkndll commented:
> Thanks for working on this. Could you re-do the migration based on the merged 17.0 version? Your migration commit is ok, but the history of the 17.0 branch you started with is a bit messy and has been cleaned-up since.

@sbidoul rebased and squashed the bot commits.

dnplkndll commented:
any update requested or this should be ready to merge

kobros-tech commented:
@lmignon 
@sbidoul 

LGTM!

sbidoul commented:
I think there is a problem with the `_authenticate` override in ir_http.py. The upstream method has changed and the arguments don't match anymore. 

I'm surprised this would work correctly.

I'd say we should rather override `_authenticate_explicit` but I'm not sure.

sbidoul reviewed (changes requested):
As mentioned in my previous comment.

sbidoul commented:
Ah, can you also migrate auth_jwt_demo in this same PR because that's where all the tests are.

kobros-tech commented:
> Ah, can you also migrate auth_jwt_demo in this same PR because that's where all the tests are.

@sbidoul 
Hi auth_jwt_demo is open now for review, https://github.com/OCA/server-auth/pull/780

sbidoul commented:
I commented on the auth_jwt_demo migration.

Have had a chance to look at https://github.com/OCA/server-auth/pull/752#issuecomment-2740815023 ?

kobros-tech commented:
@sbidoul 
thank you, but the method you commented on "_authenticate" is fixed as I guess, or you have an improvement?

grindtildeath reviewed (approved):
Looks good after testing with https://github.com/OCA/rest-framework/pull/490

sbidoul commented on auth_jwt/models/ir_http.py:
```diff
@@ -0,0 +1,143 @@
+    _inherit = "ir.http"
+
+    @classmethod
+    def _authenticate(cls, endpoint):
```
@dnplkndll I still believe we should override `_authenticate_explicit` instead. What do you think?

sbidoul commented:
Before merging this, we need a finalized auth_jwt_demo PR https://github.com/OCA/server-auth/pull/780, to make sure all tests pass.

kobros-tech commented:
@sbidoul 

I think it is very ready now?

yvaucher commented:
/ocabot merge nobump
